apps/src/applab/designElements/photochooser.jsx

```diff
@@ -0,0 +1,226 @@
+import PropertyRow from './PropertyRow';
+import BooleanPropertyRow from './BooleanPropertyRow';
+import ColorPickerPropertyRow from './ColorPickerPropertyRow';
+import ZOrderRow from './ZOrderRow';
+import EventHeaderRow from './EventHeaderRow';
+import EventRow from './EventRow';
+import BorderProperties from './BorderProperties';
+import themeValues from '../themeValues';
+import {ICON_PREFIX_REGEX} from '../constants';
+import * as elementUtils from './elementUtils';
+import * as assetPrefix from '../../assetManagement/assetPrefix';
+import designMode from '../designMode';
+import elementLibrary from './library';
+
+class PhotoChooserProperties extends React.Component {
+  static propTypes = {
+    element: PropTypes.instanceOf(HTMLElement).isRequired,
+    handleChange: PropTypes.func.isRequired,
+    onDepthChange: PropTypes.func.isRequired
+  };
+
+  handleIconColorChange = value => {
+    this.props.handleChange('icon-color', value);
+    this.props.handleChange(
+      'image',
+      this.props.element.getAttribute('data-canonical-image-url')
+    );
+  };
+
+  render() {
+    const element = this.props.element;
+
+    let iconColorPicker;
+    const canonicalImage = element.getAttribute('data-canonical-image-url');
+    if (ICON_PREFIX_REGEX.test(canonicalImage)) {
+      iconColorPicker = (
+        <ColorPickerPropertyRow
+          desc={'icon color'}
+          initialValue={elementUtils.rgb2hex(
+            element.getAttribute('data-icon-color') || '#000000'
+          )}
+          handleChange={this.handleIconColorChange}
+        />
+      );
+    }
+
+    return (
+      <div id="propertyRowContainer">
+        <PropertyRow
+          desc={'id'}
+          initialValue={elementUtils.getId(element)}
+          handleChange={this.props.handleChange.bind(this, 'id')}
+          isIdRow
+        />
+        <PropertyRow
+          desc={'width (px)'}
+          isNumber
+          initialValue={parseInt(element.style.width, 10)}
+          handleChange={this.props.handleChange.bind(this, 'style-width')}
+        />
+        <PropertyRow
+          desc={'height (px)'}
+          isNumber
+          initialValue={parseInt(element.style.height, 10)}
+          handleChange={this.props.handleChange.bind(this, 'style-height')}
+        />
+        <PropertyRow
+          desc={'x position (px)'}
+          isNumber
+          initialValue={parseInt(element.style.left, 10)}
+          handleChange={this.props.handleChange.bind(this, 'left')}
+        />
+        <PropertyRow
+          desc={'y position (px)'}
+          isNumber
+          initialValue={parseInt(element.style.top, 10)}
+          handleChange={this.props.handleChange.bind(this, 'top')}
+        />
+        <ColorPickerPropertyRow
+          desc={'background color'}
+          initialValue={elementUtils.rgb2hex(element.style.backgroundColor)}
+          handleChange={this.props.handleChange.bind(this, 'backgroundColor')}
+        />
+        {iconColorPicker}
```

Comment:
Up to you (more of a personal style preference) but you can do this inline if you'd like.
i.e.
```
      {ICON_PREFIX_REGEX.test(canonicalImage) && (
        <ColorPickerPropertyRow
          desc={'icon color'}
          initialValue={elementUtils.rgb2hex(
            element.getAttribute('data-icon-color') || '#000000'
          )}
          handleChange={this.handleIconColorChange}
        />
      )}

```